src/pytest_plugins/filler/filler.py

```diff
@@ -892,23 +888,17 @@ def pytest_collection_modifyitems(config, items):
     for item in items[:]:  # use a copy of the list, as we'll be modifying it
         if isinstance(item, EIPSpecTestItem):
             continue
-        if "fork" not in item.callspec.params or item.callspec.params["fork"] is None:
+        params: Dict[str, Any] = item.callspec.params  # type: ignore
+        if "fork" not in params or params["fork"] is None:
             items.remove(item)
             continue
-        if item.callspec.params["fork"] < Paris:
-            # Even though the `state_test` test spec does not produce an engine STATE_TEST, it does
-            # produce a BLOCKCHAIN_TEST_ENGINE, so we need to remove it here.
-            # TODO: Ideally, the logic could be contained in the `FixtureFormat` class, we create
-            # a `fork_supported` method that returns True if the fork is supported.
-            if ("state_test" in item.callspec.params) and item.callspec.params[
-                "state_test"
-            ].name.endswith("ENGINE"):
-                items.remove(item)
-            if ("blockchain_test" in item.callspec.params) and item.callspec.params[
-                "blockchain_test"
-            ].name.endswith("ENGINE"):
-                items.remove(item)
-        if "yul" in item.fixturenames:
+        fork: Fork = params["fork"]
+        if fork.engine_new_payload_version(0, 0) is None:
```

Comment:
Just a small helper for the reader.
```suggestion
        # don't generate an "engine" fixture variation pre-merge forks
        if fork.engine_new_payload_version(0, 0) is None:
```

Reply:
I think we can actually even do something like `fixture_type.supports_fork(fork)` so this is abstracted a bit better.

I'll give it a 5-min try.

Reply:
> I'll give it a 5-min try.

lmao this prompted a huge refactor.